### PR TITLE
Undeprecate int constants for interpolation

### LINKED ISSUE #7205: Revert removal and deprecation of int values for interpolation?
Initially, `torchvision` supported `int`'s for the `interpolation` (or their former equivalent) arguments on our transforms. They were usually not passed directly, but rather as part of the Pillow constants, like `PIL.Image.BILINEAR`.

https://github.com/pytorch/vision/pull/2952#pullrequestreview-524098958 proposed to decouple our transforms from Pillow and this was achieved through the `torchvision.transforms.InterpolationMode` enum introduced later in the same PR. Although, the new enum was used as default value from then on, `int`'s were still supported. Still, the language around them somewhat "soft-deprecated" them:

https://github.com/pytorch/vision/blob/9b5a3fecc72434dbd65148723efe54b28b9728c9/torchvision/transforms/transforms.py#L301

With the release of `Pillow==9.1.0`, they [deprecated the `int` constants](https://pillow.readthedocs.io/en/stable/releasenotes/9.1.0.html#constants) and switched to their own enums as well. In response we introduced support for these enums as well in #5898 and deprecated `int`s in #5974.

Our deprecation was scheduled for removal in the upcoming 0.15 release and was acted upon in #7176.

However, there are two factors we didn't consider there:

1. @NicolasHug pointed out that using ints is still widespread for Meta internal users
2. `Pillow` seemed to have [walked back on their deprecation](https://pillow.readthedocs.io/en/stable/releasenotes/9.4.0.html#constants) in `9.4.0` as well:

   > In Pillow 9.1.0, the following constants were deprecated. That has been reversed and these constants will now remain available.

   This follows python-pillow/Pillow#6537 which acknowledges that the change is too disruptive. However, since removing the enums now would also be BC breaking (python-pillow/Pillow#6684), they just reinstated the former int constants in python-pillow/Pillow@bcdb208fe2365b75c8d87d29783b9cc9f8cb683b

Given that our deprecation just followed `Pillow`'s, I think we can also revert the removal and deprecation. Between `>= 9.1.0` and `< 9.4.0`, `Pillow` never dropped support for `int`'s. Thus, the only issues our users could run into when using `int`'s on that versions is that `Pillow` will emit a deprecation warning. Our default values use our enum anyway

https://github.com/pytorch/vision/blob/539c6e290c172d173f986874d57be857235fbc6a/torchvision/transforms/transforms.py#L315

If we go that route note that we also need add support for `int`'s to our prototype transforms. They don't have that, since we designed with the scheduled removal in mind.

Thoughts?

cc @vfdev-5

### PR DESCRIPTION
**Pull request by pmeier**
Closes #7205. TL;DR this PR undoes #7176, #5974, #5898 for the integer inputs for interpolation mode. They are now fully supported again and no longer emit a deprecation warning. I'll send a follow-up PR to add this functionality to the prototype transforms as well.

cc @vfdev-5

### CODE REVIEW

File: test/test_functional_tensor.py [hunk 1]
```diff
@@ -144,6 +144,12 @@ def test_rotate_batch(self, device, dt):
         center = (20, 22)
         _test_fn_on_batch(batch_tensors, F.rotate, angle=32, interpolation=NEAREST, expand=True, center=center)
 
+    def test_rotate_interpolation_type(self):
+        tensor, _ = _create_data(26, 26)
+        res1 = F.rotate(tensor, 45, interpolation=2)
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:149]: should we use `PIL.Image.BILINEAR` instead?
(here and in other places)

File: test/test_functional_tensor.py [hunk 2]
```diff
@@ -489,9 +513,7 @@ def test_resize(device, dt, size, max_size, interpolation):
 
     assert resized_tensor.size()[1:] == resized_pil_img.size[::-1]
 
-    if interpolation not in [
-        NEAREST,
-    ]:
+    if interpolation not in [NEAREST]:
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:516]: nit

```suggestion
    if interpolation != NEAREST:
```

File: torchvision/transforms/functional.py [hunk 3]
```diff
@@ -454,8 +455,12 @@ def resize(
     if not torch.jit.is_scripting() and not torch.jit.is_tracing():
         _log_api_usage_once(resize)
 
-    if not isinstance(interpolation, InterpolationMode):
-        raise TypeError("Argument interpolation should be a InterpolationMode")
+    if isinstance(interpolation, int):
+        interpolation = _interpolation_modes_from_int(interpolation)
+    elif not isinstance(interpolation, InterpolationMode):
+        raise TypeError(
+            "Argument interpolation should be a InterpolationMode or an corresponding Pillow integer constant"
```
Comment (NicolasHug) [inline torchvision/transforms/functional.py:462]: (same in similar error messages)

```suggestion
            "Argument interpolation should be a InterpolationMode or a corresponding Pillow integer constant"
            f"Got {interpolation} of type {type(interpolation).}"
```

**Review by NicolasHug — COMMENTED** [review]
Some first round but this looks good. According to the failing test is looks like we still have some remnants of `_pil_constants.py` in a few places

File: test/test_functional_tensor.py (hunk 1 above)
Comment (pmeier) [reply test/test_functional_tensor.py:149]: I was just putting back the original content, but happy to change to the named constants if that is ok with you.

**Review by NicolasHug — APPROVED** [review]
Thanks a lot Philip.

Do we.. need to bring back support for it on the v2 transforms as well? Let's mark it as "nice to have" on https://github.com/pytorch/vision/issues/7217?

**Comment by vfdev-5** [comment]
@pmeier this PR brings back warnings like
```
  /vision/torchvision/transforms/functional_pil.py:286: DeprecationWarning: NEAREST is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.NEAREST or Dither.NONE instead.
    interpolation: int = Image.NEAREST,
```

**Pull request merged**